tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -107,6 +107,32 @@ def ldap_tear_down():
         ldap_auth.delete()
 
 
+@fixture()
+def groups_teardown():
+    "teardown for groups created for external/remote groups"
+    yield
+    # tier down groups
+    for group_name in ('sat_users', 'sat_admins'):
+        user_groups = entities.UserGroup().search(query={'search': f'name="{group_name}"'})
+        if user_groups:
+            user_groups[0].delete()
+
+
+@fixture()
+def multigroup_setting_cleanup():
+    "Adding and removing the user to/from ipa group"
+    sat_admins = settings.ipa.groups.replace(" ", '').split(',')[0]
+    idm_user = settings.ipa.group_users.replace(" ", '').split(',')[1]
```

Comment:
If these values are comma-separated list of values, they should have been casted to list during reading in configuration module.

I don't like that we perform cleanup and processing on test level, and I'm not huge fan of hardcoded magic values.
We control robottelo.properties file - why did we put value that is harder to work with on test level?
If we didn't - what exactly are we guarding against here?
I would say we shouldn't be too defensive. And when we do, these guards are better implemented in some helper or filter than in test directly.

Reply:
Do you mean updating the structure in base.py, so it will be presented as list?

Reply:
Yes, our `reader.get()` has ability to cast value to specific type:
```python
self.groups = reader.get('ipa', 'groups', cast=list)
```
It might be good idea to also add `default=[]`, or whatever value makes sense here.